main.go

```diff
@@ -118,8 +165,27 @@ func main() {
 		}
 	})
 
-	// send klog messages through our zap logger so they look the same (console vs JSON)
-	klog.SetLogger(logr)
+	// Prepare a zap configuration for klog, so that all the logs look similar.
+
+	if zapEncoder == "console" { // klog already adds a newline, so have zap skip adding another in console mode
+		zapcfg.EncoderConfig.SkipLineEnding = true
+	}
+
+	klogLevel, err := strconv.Atoi(pflag.Lookup("v").Value.String())
+	if err != nil {
```

Comment:
Is it an error if I don't specify `-v` at all? I wouldn't want an error log message due to that is all.

Reply:
Not specifying it will not cause an error. Actually I think this error case should be impossible to hit (I think klog will ensure it's valid when it parses the arguments), but it was easy enough to handle the error case as opposed to telling the linter and sonarcloud that we can skip the error check.